### ISSUE TITLE
Failing to run jrnl ModuleNotFoundError: No module named 'pkg_resources'

### ISSUE THREAD

**Issue by galloramiro** [jrnl-org/jrnl#1950]
### Diagnostic output

```sh
❯ jrnl --diagnostic
Traceback (most recent call last):
  File "/home/blabla/.local/bin/jrnl", line 5, in <module>
    from jrnl.cli import run
  File "/home/blabla/.local/share/pipx/venvs/jrnl/lib64/python3.13/site-packages/jrnl/__init__.py", line 4, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

### Current Behavior

Trying to run any command and just get the same error

### Expected Behavior

Able to run commands

### Repro Steps

jrnl any jrnl command

### Debug output

```bash
❯ jrnl --diagnostic --debug
Traceback (most recent call last):
  File "/home/blabla/.local/bin/jrnl", line 5, in <module>
    from jrnl.cli import run
  File "/home/blabla/.local/share/pipx/venvs/jrnl/lib64/python3.13/site-packages/jrnl/__init__.py", line 4, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

### Other Information

```bash
❯ python --version
Python 3.13.0
```
![Screenshot_20241104_092450](https://github.com/user-attachments/assets/2a59c5b1-063b-41cd-a97c-4f049657a65e)

**Comment by galloramiro** [comment]
Fix this doing the following command.
```bash
pipx inject jrnl setuptools
```
but now it's not recognizing the commands 
```bash
jrnl: error: unrecognized arguments: --config-file --template /blabla/jrnl/templates/day_start.md
```

**Comment by galloramiro** [comment]
I just list the instalation and realize that was a pretty old version... 

```bash
❯ pipx list
venvs are in /home/blabla/.local/share/pipx/venvs
apps are exposed on your $PATH at /home/blabla/.local/bin
manual pages are exposed at /home/blabla/.local/share/man
   package jrnl 2.1.post2, installed using Python 3.13.0
    - jrnl
```

I proceed to uninstall it 
```bash
❯ pipx uninstall jrnl
uninstalled jrnl! ✨ 🌟 ✨
❯ pipx list
nothing has been installed with pipx 😴
```

Finally try to installit again, and this is what i get:
```bash
❯ pipx install jrnl==4.1
Fatal error from pip prevented installation. Full pip output in file:
    /home/blabla/.local/state/pipx/log/cmd_2024-11-06_10.06.07_pip_errors.log

Some possibly relevant errors from pip install:
    ERROR: Ignored the following versions that require a different python version: 2.1.1 Requires-Python >=3.6.0,<3.8.0; 2.1.1b0 Requires-Python >=3.6.0,<3.8.0; 2.2 Requires-Python >=3.6.0,<3.9.0; 2.2.1b2 Requires-Python >=3.6.0,<3.9.0; 2.2b0 Requires-Python >=3.6.0,<3.9.0; 2.2b2 Requires-Python >=3.6.0,<3.9.0; 2.3 Requires-Python >=3.6.0,<3.9.0; 2.3.1 Requires-Python >=3.6.0,<3.9.0; 2.3.1b0 Requires-Python >=3.6.0,<3.9.0; 2.3b0 Requires-Python >=3.6.0,<3.9.0; 2.4 Requires-Python >=3.6.0,<3.9.0; 2.4.1 Requires-Python >=3.6.0,<3.9.0; 2.4.2 Requires-Python >=3.6.0,<3.9.0; 2.4.3 Requires-Python >=3.6.0,<3.9.0; 2.4.3b0 Requires-Python >=3.6.0,<3.9.0; 2.4.4 Requires-Python >=3.6.0,<3.9.0; 2.4.4b0 Requires-Python >=3.6.0,<3.9.0; 2.4.5 Requires-Python >=3.6.0,<3.9.0; 2.4b0 Requires-Python >=3.6.0,<3.9.0; 2.5 Requires-Python >=3.7.0,<3.10.0; 2.5.1a0 Requires-Python >=3.7.0,<3.10.0; 2.5b0 Requires-Python >=3.7.0,<3.10.0; 2.6 Requires-Python >=3.7.0,<3.10; 2.6b0 Requires-Python >=3.7.0,<3.10; 2.6b2 Requires-Python >=3.7.0,<3.10; 2.6b3 Requires-Python >=3.7.0,<3.10; 2.6b4 Requires-Python >=3.7.0,<3.10; 2.6b5 Requires-Python >=3.7.0,<3.10; 2.7 Requires-Python >=3.7.0,<3.10; 2.7.1 Requires-Python >=3.7.0,<3.10; 2.7.1b0 Requires-Python >=3.7.0,<3.10; 2.7.2b0 Requires-Python >=3.7.0,<3.10; 2.7b0 Requires-Python >=3.7.0,<3.10; 2.8 Requires-Python >=3.7.0,<3.10; 2.8.1 Requires-Python >=3.7.0,<3.10; 2.8.1b0 Requires-Python >=3.7.0,<3.10; 2.8.2 Requires-Python >=3.7.0,<3.11; 2.8.2b0 Requires-Python >=3.7.0,<3.11; 2.8.3 Requires-Python >=3.7.0,<3.11; 2.8.3b0 Requires-Python >=3.7.0,<3.11; 2.8.4 Requires-Python >=3.7.0,<3.12; 2.8.4b0 Requires-Python >=3.7.0,<3.12; 2.8.4b1 Requires-Python >=3.7.0,<3.12; 2.8.4b2 Requires-Python >=3.7.0,<3.12; 2.8b0 Requires-Python >=3.7.0,<3.10; 3.0 Requires-Python >=3.9.0,<3.12; 3.0b0 Requires-Python >=3.9.0,<3.12; 3.0b2 Requires-Python >=3.9.0,<3.12; 3.1 Requires-Python >=3.9.0,<3.12; 3.1b0 Requires-Python >=3.9.0,<3.12; 3.1b2 Requires-Python >=3.9.0,<3.12; 3.2 Requires-Python >=3.9.0,<3.12; 3.2b0 Requires-Python >=3.9.0,<3.12; 3.3 Requires-Python >=3.9.0,<3.12; 3.3b0 Requires-Python >=3.9.0,<3.12; 3.3b2 Requires-Python >=3.9.0,<3.12; 4.0 Requires-Python >=3.10.0,<3.13; 4.0.1 Requires-Python >=3.10.0,<3.13; 4.0.1b0 Requires-Python >=3.10.0,<3.13; 4.0b0 Requires-Python >=3.10.0,<3.13; 4.0b2 Requires-Python >=3.10.0,<3.13; 4.0b3 Requires-Python >=3.10.0,<3.13; 4.1 Requires-Python >=3.10.0,<3.13; 4.1b0 Requires-Python >=3.10.0,<3.13; 4.1b2 Requires-Python >=3.10.0,<3.13
    ERROR: Could not find a version that satisfies the requirement jrnl==4.1 (from versions: 0.3.0, 1.0.0, 1.5.5, 1.5.6, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.7.5, 1.7.6, 1.7.7, 1.7.8, 1.7.9, 1.7.10, 1.7.12, 1.7.13, 1.7.14, 1.7.15, 1.7.16, 1.7.17, 1.7.18, 1.7.19, 1.7.20, 1.7.22, 1.8.0, 1.8.1, 1.8.4, 1.8.6, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.9.5, 1.9.6, 1.9.7, 1.9.8, 2.0.0, 2.0.1b1, 2.0.1, 2.1b2, 2.1b3, 2.1b4, 2.1b6, 2.1.post2, 4.2b0)
    ERROR: No matching distribution found for jrnl==4.1

Error installing jrnl from spec 'jrnl==4.1'.
```

**Comment by galloramiro** [comment]
Finally fix it by installing it with dnf instead of pipx

**Comment by micahellison** [comment]
Hi @galloramiro, sorry for missing this, and thanks for documenting your findings here. This issue shouldn't be a problem in pipx anymore -- the issue was that you're running the very new Python 3.13, and the pipx version solver knew that the latest version of jrnl didn't support Python 3.13, so it went back to an ancient version that had an open-ended upper Python version.

Now that we've released jrnl 4.2 with explicit Python 3.13 support, this should no longer be a problem when installing through pipx.

We usually update just in time for the latest Python version, but we're a bit late this year. I'll keep this in mind when 3.14 rolls around next year.

**Comment by galloramiro** [comment]
Thanks. I suppose something like that, but wasn't sure when I was trying to debug it hahaha
